docs: add minimal docs for self-managed infrastructure (#2458)
edgelesssys/constellation: docs/docs/workflows/create.md

```bash
CONSTELL_IP=$(terraform output ip)

> CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)

Author: For the internal LB feature, I assume?

> CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)

Author: We should probably make sure that the TF output names are aligned with the state file fields so that we can just refer to `terraform output` here and the user can trivially map them to the state file fields. Would save some explanation here

> CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)

Reviewer: This is still not up to date. The new output field are named `out_of_cluster_endpoint` and `in_cluster_endpoint`.

> CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)

Reviewer: I don't know how trivially we can change the `Infrastructure` section of the state file or if we need a migration. I'm fine either way you want to align them.

> CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)

Reviewer: They are already in the state file (see main branch) but be also need to document here, right?

yq eval '.infrastructure.initSecret ="$CONSTELL_INIT_SECRET"' --inplace constellation-state.yaml
yq eval '.infrastructure.clusterEndpoint ="$CONSTELL_IP"' --inplace constellation-state.yaml
```

</tabItem>

> <tabItem value="self-managed" label="Self-Managed">

Reviewer 2: ```suggestion
<tabItem value="self-managed" label="Self-managed">
```

Self-managed infrastructure allows for managing the cloud resources necessary for a Constellation cluster separate from the Constellation CLI.

> This provides maximum flexibility in DevOps and can meet potential regulatory requirements.

Reviewer 2: ```suggestion
This provides flexibility in DevOps and can meet potential regulatory requirements.
```

To self-manage the infrastructure of your cluster, download the Terraform files for the selected CSP from the [Constellation GitHub repository](https://github.com/edgelesssys/constellation/tree/main/cli/internal/terraform/terraform).
They contain a minimum configuration for the resources necessary to run a Constellation cluster on the corresponding CSP. From this base, you can now add, edit, or substitute resources per your own requirements with the infrastructure

> management tooling of your choice. You need to ensure the essential functionality of the base configuration is kept in order for your cluster to function correctly.

Reviewer 2: ```suggestion
management tooling of your choice. You need to keep the essential functionality of the base configuration in order for your cluster to function correctly.
```

> Make sure all necessary resources are created, e.g. through checking your CSP's portal and retrieve the necessary values, aligned with the outputs (specified in `outputs.tf`) of the base configuration.

Reviewer 2: ```suggestion
Make sure all necessary resources are created, e.g., through checking your CSP's portal and retrieve the necessary values, aligned with the outputs (specified in `outputs.tf`) of the base configuration.
```

```bash
CONSTELL_IP=$(terraform output ip)
CONSTELL_INIT_SECRET=$(terraform output initSecret | jq -r | tr -d '\n' | base64)
yq eval '.infrastructure.initSecret ="$CONSTELL_INIT_SECRET"' --inplace constellation-state.yaml
yq eval '.infrastructure.clusterEndpoint ="$CONSTELL_IP"' --inplace constellation-state.yaml
```

</tabItem>
<tabItem value="self-managed" label="Self-Managed">

> Self-managed infrastructure allows for managing the cloud resources necessary for a Constellation cluster separately from the Constellation CLI and Terraform,

Reviewer 3: ```suggestion
Self-managed infrastructure allows for managing the cloud resources necessary for a Constellation cluster separately from the Constellation CLI,
```
Our default workflow does not directly include Terraform, but a user might wish to use Terraform to self-manage the resources.

providing maximum flexibility in DevOps as well as meeting potential regulatory requirements.

To self-manage the infrastructure of your cluster, download the Terraform files for the selected CSP from the [Constellation GitHub repository](https://github.com/edgelesssys/constellation/tree/main/cli/internal/terraform/terraform).
They contain a minimum configuration for the resources necessary to run a Constellation cluster on the corresponding CSP. From this base, you can now add, edit, or substitute resources as per your own requirements, while ensuring the essential
functionality of the base configuration is kept. You can also recreate the infrastructure from the Terraform configuration with the infrastructure management tooling of your choice, e.g. create it manually through a CSP portal.

> When using Terraform, initialize and apply your configuration. Otherwise, make sure all necessary resources are created, e.g. through checking your CSP's portal.

Reviewer 3: Maybe having the steps required when using Terraform and steps for different solutions in tabs, similar how we document different CSPs, would clear up this section slightly.

e.g. the default would show what to do when using Terraform, and a separate tab would tell the user to go through the CSP portal to create/check resources.
This would also allow us to more easily add new options in the future.

```bash
terraform init
terraform apply
```

> And retrieve the necessary outputs from your configuration. When using another infrastructure management strategy than Terraform, retrieve the outputs accordingly, e.g. through your CSP's portal.

Reviewer 3: ```suggestion
Retrieve the necessary outputs from your configuration. When using another infrastructure management strategy than Terraform, retrieve the outputs accordingly, e.g. through your CSP's portal.
```

```bash
CONSTELL_IP=$(terraform output ip)

> CONSTELL_INIT_SECRET=$(terraform output initSecret | jq -r | tr -d '\n' | base64)

Reviewer 3: ```suggestion
CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)
```

yq eval '.infrastructure.initSecret ="$CONSTELL_INIT_SECRET"' --inplace constellation-state.yaml
yq eval '.infrastructure.clusterEndpoint ="$CONSTELL_IP"' --inplace constellation-state.yaml
```

Depending on the [configuration](./config.md) of your cluster, additional outputs might be necessary. Use the following command to query all possible outputs:

```bash
terraform output

> ```

Reviewer 3: Maybe having just this section with `terraform output` would be clearer for the user, instead of having partial commands to fill some of the statefile.
If we aim to keep statefile field names and output field names as close as possible and/or document the fields extensively, it should be easy to fill in the required values without much additional documentation here.

> And continue with [initializing your cluster](#the-init-step).

Reviewer 3: ```suggestion
Continue with [initializing your cluster](#the-init-step).
```

```bash
CONSTELL_IP=$(terraform output ip)

> CONSTELL_INIT_SECRET=$(terraform output initSecret | yq -r | tr -d '\n' | base64)

Reviewer: I need to leave a comment to request changes again. So this is a reference to https://github.com/edgelesssys/constellation/pull/2458/files#r1362159049.
